dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-dameng/src/main/java/org/apache/dolphinscheduler/plugin/datasource/dameng/param/DamengDataSourceProcessor.java

```diff
@@ -0,0 +1,160 @@
+import java.sql.SQLException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.service.AutoService;
+
+@AutoService(DataSourceProcessor.class)
+public class DamengDataSourceProcessor extends AbstractDataSourceProcessor {
+
+    private final Logger logger = LoggerFactory.getLogger(DamengDataSourceProcessor.class);
+
+    @Override
+    public BaseDataSourceParamDTO castDatasourceParamDTO(String paramJson) {
+        return JSONUtils.parseObject(paramJson, DamengDataSourceParamDTO.class);
+    }
+
+    @Override
+    public BaseDataSourceParamDTO createDatasourceParamDTO(String connectionJson) {
+        DamengConnectionParam connectionParams = (DamengConnectionParam) createConnectionParams(connectionJson);
+        DamengDataSourceParamDTO damengDatasourceParamDTO = new DamengDataSourceParamDTO();
+
+        damengDatasourceParamDTO.setUserName(connectionParams.getUser());
+        damengDatasourceParamDTO.setDatabase(connectionParams.getDatabase());
+        damengDatasourceParamDTO.setOther(parseOther(connectionParams.getOther()));
+
+        String address = connectionParams.getAddress();
+        String[] hostSeperator = address.split(Constants.DOUBLE_SLASH);
+        String[] hostPortArray = hostSeperator[hostSeperator.length - 1].split(Constants.COMMA);
+        damengDatasourceParamDTO.setPort(Integer.parseInt(hostPortArray[0].split(Constants.COLON)[1]));
+        damengDatasourceParamDTO.setHost(hostPortArray[0].split(Constants.COLON)[0]);
+
+        return damengDatasourceParamDTO;
+    }
+
+    @Override
+    public BaseConnectionParam createConnectionParams(BaseDataSourceParamDTO dataSourceParam) {
+        DamengDataSourceParamDTO dmDatasourceParam = (DamengDataSourceParamDTO) dataSourceParam;
+        String address = String
+                .format("%s%s:%s", DataSourceConstants.JDBC_DAMENG, dmDatasourceParam.getHost(),
+                        dmDatasourceParam.getPort());
+        String jdbcUrl = StringUtils.isEmpty(dmDatasourceParam.getDatabase()) ? address
+                : String.format("%s/%s", address,
+                        dmDatasourceParam.getDatabase());
+
+        DamengConnectionParam damengConnectionParam = new DamengConnectionParam();
+        damengConnectionParam.setJdbcUrl(jdbcUrl);
+        damengConnectionParam.setDatabase(damengConnectionParam.getDatabase());
+        damengConnectionParam.setAddress(address);
+        damengConnectionParam.setUser(dmDatasourceParam.getUserName());
+        damengConnectionParam.setPassword(PasswordUtils.encodePassword(dmDatasourceParam.getPassword()));
+        damengConnectionParam.setDriverClassName(getDatasourceDriver());
+        damengConnectionParam.setValidationQuery(getValidationQuery());
+        damengConnectionParam.setOther(transformOther(dmDatasourceParam.getOther()));
+        damengConnectionParam.setProps(dmDatasourceParam.getOther());
```

Comment:
Please don't use setProps, I remember we don't have props in `BaseDataSourceParamDTO`

Reply:
Ok, I have rebase BaseDataSourceParamDTO